nsv13/code/modules/munitions/ship_weapons/energy_weapons/subspacecooling.dm

```diff
@@ -0,0 +1,107 @@
+/obj/machinery/cooling
+	name = "subspace unit"
+	desc = "A subspace unit."
+	icon = 'nsv13/icons/obj/subspace.dmi'
+	icon_state = "cooler"
+	circuit = /obj/item/circuitboard/machine
+	bound_width = 32
+	pixel_x = 0
+	pixel_y = 0
+	idle_power_usage =  2000
+	var/obj/machinery/ship_weapon/energy/parent
+	var/on = FALSE
+	density = TRUE
+	critical_machine = TRUE
+
+/obj/machinery/cooling/cooler
+	name = "subspace cooling unit"
+	desc = "A cooling unit that dumps the massive amounts of heat energy weapons generate into subspace."
+	circuit = /obj/item/circuitboard/machine/cooling
+
+
+
+/obj/machinery/cooling/cooler/Initialize(mapload)
+	. = ..()
+	var/obj/machinery/ship_weapon/energy/E = locate(/obj/machinery/ship_weapon/energy) in orange(1, src) //I have no idea what I'm doing and this causes errors so
+	E.coolers |= src
+	parent = E
+
+/obj/machinery/cooling/storage/Initialize(mapload)
+	. = ..()
+	var/obj/machinery/ship_weapon/energy/E = locate(/obj/machinery/ship_weapon/energy) in orange(1, src)
+	E.storages |= src
+	parent = E
+
+
+/obj/machinery/cooling/cooler/Destroy()
+  parent.coolers -= src
+  . = ..()
+
+/obj/machinery/cooling/storage/Destroy()
+  parent.storages -= src
+  . = ..()
+
+/obj/machinery/cooling/attack_hand(mob/user)
+	. = ..()
+	if(panel_open)
+		to_chat(user, "<span class='notice'>You must turn close the panel on [src] before turning it on.</span>")
+		return
+	to_chat(user, "<span class='notice'>You press [src]'s power button.</span>")
+	on = !on
+	update_icon()
+
+/obj/machinery/cooling/examine()
+	. = ..()
+	if(on)
+		. += "The power is on"
+	if(parent)
+		. += "The thermal subspace transcever is linked"
+	else
+		. += "it's completely inactive"
+
+
+/obj/machinery/cooling/update_icon()
+	cut_overlays()
+	if(panel_open)
+//		icon_state = "smes-o"
+	if(on && parent)
+//		add_overlay("smes-op1")
+	if(on)
+//		add_overlay("smes-oc1")
+	else
+//		add_overlay("smes-op0")
+
+
+/obj/machinery/cooling/multitool_act(mob/living/user, obj/item/multitool/I)
```

Comment:
This proc does not check whether this machine is already linked, which allows you to link one cooler to several weapons.

Reply:
Aw shoot. I've got to fix that. I completely didn't think of that

Reply:
So maybe do parent.coolers = null and then parent = null before handling adding itself to another weapon?

Reply:
Or alternatively, if parent.coolers = src then return false?

Reply:
>So maybe do parent.coolers = null and then parent = null before handling adding itself to another weapon?

Yesn't. You would have to remove it from the list, however *nulling* `parent.coolers` would drop the *entire* list (all coolers, also the fact that variable was a list). See list operators for removal of a specific element (`- / -=` removal / `+ / +=` add)

Reply:
yeah that was my brain being stupid, I saw that just now- I want the same handling as in destroy, parent.coolers =- src or however it goes